src/core/onerror.js

```diff
@@ -21,6 +23,8 @@ import onUncaughtException from "./on-uncaught-exception";
  * @return {bool} True if native error reporting should be suppressed.
  */
 export default function onWindowError( details ) {
+	Logger.warn( "QUnit.onError is deprecated and will be removed in QUnit 3.0." );
```

Comment:
Is there a suggested fix?
Use `QUnit.on('error', ...)`?

Reply:
No, I did not provide a direct replacement. This was an internal method in my mind, with a rather awkward/incompete responsibility. It wasn't exactly a `window.onerror` callback (because we had additional logic in our actual window.onerror callback in html.js, before we hit this function), but it also wasn't exactly a "push global failure" utility.

As part of 07de3c6fdea60c0a5279eefebb30002171dafcda, I factored out what I thought was a sensible re-usable utility into `QUnit.onUnhandledRejection()`, and documented it. And in this PR, I copied the remaining logic from here to the actual window handler in `html.js`.

If someone out there did end up using this, as part of a custom HTML Reporter or something, and consumed this method, then a replacement would involve calling `onUncaughtException()`, plus whatever else this method currently does. I don't expect it to be likely, but it's something worth clarifying in the warning message, and to call out in the upcoming migration guide.

Speaking of migration guide, I've made a note of that at https://github.com/qunitjs/qunit/issues/1498.

TODO: Improve console warning message.